pkg/lib/harness/llm-harness.go

```diff
@@ -181,8 +183,26 @@ func PrintLogs(configHome string, w io.Writer) error {
 		return fmt.Errorf("error reading log file: %w", err)
 	}
 	defer logFile.Close()
-	if _, err = io.Copy(w, logFile); err != nil {
-		return fmt.Errorf("failed to print log file: %w", err)
+	if !tail {
+		if _, err = io.Copy(w, logFile); err != nil {
+			return fmt.Errorf("failed to print log file: %w", err)
+		}
```

Comment:
You can simplify early returning in here.  Alternatively you can handle the `!tail` case in the loop.
```
return nil
```